test/trace_test/main.cpp

```diff
@@ -0,0 +1,51 @@
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+#include "../../src/traces/traces.hpp"
+
+/// Print out two traces
+int main() {
+    std::vector<Trace> traces;
+    std::vector<Trace> random_oracle = {
```

Comment:
```suggestion
    std::array<Trace, trace_count> random_oracle = {
```
Compiler seems to not like the initialiser list. It can't infer the right constructor because it can't infer the element type. A workaround is to use std::array which only has 1 constructor. But to use std::array, we need to give it a static size.